etc/scripts/utils_thirdparty.py

```diff
@@ -0,0 +1,2962 @@
+        notice_file = about_data.pop('notice_file', None)
+        if notice_text:
+            about_data['notice_text'] = notice_text
+        elif notice_file:
+            notice_loc = os.path.join(dest_dir, notice_file)
+            if os.path.exists(notice_loc):
+                with open(notice_loc) as fi:
+                    about_data['notice_text'] = fi.read()
+        return self.update(about_data, keep_extra=True)
+
+    def load_remote_about_data(self):
+        """
+        Fetch and update self with "remote" data Distribution ABOUT file and
+        NOTICE file if any. Return True if the data was updated.
+        """
+        try:
+            about_text = fetch_content_from_path_or_url_through_cache(self.about_download_url)
+        except RemoteNotFetchedException:
+            return False
+
+        if not about_text:
+            return False
+
+        about_data = saneyaml.load(about_text)
+        notice_file = about_data.pop('notice_file', None)
+        if notice_file:
+            try:
+                notice_text = fetch_content_from_path_or_url_through_cache(self.notice_download_url)
+                if notice_text:
+                    about_data['notice_text'] = notice_text
+            except RemoteNotFetchedException:
+                print(f'Failed to fetch NOTICE file: {self.notice_download_url}')
+        return self.load_about_data(about_data)
+
+    def get_checksums(self, dest_dir=THIRDPARTY_DIR):
+        """
+        Return a mapping of computed checksums for this dist filename is
+        `dest_dir`.
+        """
+        dist_loc = os.path.join(dest_dir, self.filename)
+        if os.path.exists(dist_loc):
+            return multi_checksums(dist_loc, checksum_names=('md5', 'sha1', 'sha256'))
+        else:
+            return {}
+
+    def set_checksums(self, dest_dir=THIRDPARTY_DIR):
+        """
+        Update self with checksums computed for this dist filename is `dest_dir`.
+        """
+        self.update(self.get_checksums(dest_dir), overwrite=True)
+
+    def validate_checksums(self, dest_dir=THIRDPARTY_DIR):
+        """
+        Return True if all checksums that have a value in this dist match
+        checksums computed for this dist filename is `dest_dir`.
+        """
+        real_checksums = self.get_checksums(dest_dir)
+        for csk in ('md5', 'sha1', 'sha256'):
+            csv = getattr(self, csk)
+            rcv = real_checksums.get(csk)
+            if csv and rcv and csv != rcv:
+                return False
+        return True
+
+    def get_pip_hash(self):
+        """
+        Return a pip hash option string as used in requirements for this dist.
+        """
+        assert self.sha256, f'Missinh SHA256 for dist {self}'
+        return f'--hash=sha256:{self.sha256}'
+
+    def get_license_keys(self):
+        try:
+            keys = LICENSING.license_keys(self.license_expression, unique=True, simple=True)
+        except license_expression.ExpressionParseError:
```

Comment:
@pombredanne 

I ran into an issue with the license_expression library throwing an `ExpressionParseError` when I ran the `bootstrap.py` script without `scancode-toolkit` installed. The license expression for this package was the classifier strings that contain the license. The LICENSING object would balk at the symbol `License`. My initial reaction to this is to return a list that only contains `unknown`, but I am not sure if this would be the best thing to do.